File: scripts/deploy-pages.js
const fs = require('fs');
const path = require('path');
const https = require('https');

const ACCOUNT_ID = 'dfbee5c2a5706a81bc04675499c933d4';
const PROJECT_NAME = 'upay-client-kc';
const TOKEN = 'cfut_U9fdeBzWaVKBK83pfPZAtmufVQnJqtCTKLQj53UFad7fbeca';
const SOURCE_DIR = path.join(__dirname, '..', 'UpayClient', '_KC');

// 生成 manifest
function generateManifest(dir) {
    const files = {};
    const items = fs.readdirSync(dir);
    
    for (const item of items) {
        const fullPath = path.join(dir, item);
        const stat = fs.statSync(fullPath);
        
        if (stat.isFile()) {
            const content = fs.readFileSync(fullPath);
            const hash = require('crypto').createHash('sha256').update(content).digest('hex');
            files[item] = {
                hash: hash,
                size: stat.size
            };
        }
    }
    
    return { manifest: { files } };
}

// 創建 multipart form-data
function createFormData(manifest, files) {
    const boundary = '----FormBoundary' + Math.random().toString(36).substring(2);
    const chunks = [];
    
    // Add manifest
    chunks.push(Buffer.from(`--${boundary}\r\n`));
    chunks.push(Buffer.from('Content-Disposition: form-data; name="manifest"\r\n\r\n'));
    chunks.push(Buffer.from(JSON.stringify(manifest)));
    chunks.push(Buffer.from('\r\n'));
    
    // Add files
    for (const [filename, filepath] of Object.entries(files)) {
        const content = fs.readFileSync(filepath);
        chunks.push(Buffer.from(`--${boundary}\r\n`));
        chunks.push(Buffer.from(`Content-Disposition: form-data; name="${filename}"; filename="${filename}"\r\n`));
        chunks.push(Buffer.from('Content-Type: application/octet-stream\r\n\r\n'));
        chunks.push(content);
        chunks.push(Buffer.from('\r\n'));
    }
    
    chunks.push(Buffer.from(`--${boundary}--\r\n`));
    
    return {
        buffer: Buffer.concat(chunks),
        boundary
    };
}

async function deploy() {
    console.log('📦 Generating manifest...');
    const manifest = generateManifest(SOURCE_DIR);
    
    const files = {};
    const items = fs.readdirSync(SOURCE_DIR);
    for (const item of items) {
        const fullPath = path.join(SOURCE_DIR, item);
        if (fs.statSync(fullPath).isFile()) {
            files[item] = fullPath;
        }
    }
    
    console.log(`📁 Files to deploy: ${Object.keys(files).length}`);
    
    const { buffer, boundary } = createFormData(manifest, files);
    
    const options = {
        hostname: 'api.cloudflare.com',
        port: 443,
        path: `/client/v4/accounts/${ACCOUNT_ID}/pages/projects/${PROJECT_NAME}/deployments`,
        method: 'POST',
        headers: {
            'Authorization': `Bearer ${TOKEN}`,
            'Content-Type': `multipart/form-data; boundary=${boundary}`,
            'Content-Length': buffer.length
        }
    };
    
    console.log('🚀 Uploading to Cloudflare Pages...');
    
    return new Promise((resolve, reject) => {
        const req = https.request(options, (res) => {
            let data = '';
            res.on('data', (chunk) => data += chunk);
            res.on('end', () => {
                try {
                    const result = JSON.parse(data);
                    if (result.success) {
                        console.log('✅ Deployment successful!');
                        console.log(`🌐 URL: https://${PROJECT_NAME}.pages.dev`);
                        resolve(result);
                    } else {
                        console.error('❌ Deployment failed:', result.errors);
                        reject(result.errors);
                    }
                } catch (e) {
                    console.error('❌ Parse error:', data);
                    reject(e);
                }
            });
        });
        
        req.on('error', (err) => {
            console.error('❌ Request error:', err);
            reject(err);
        });
        
        req.write(buffer);
        req.end();
    });
}

deploy().catch(console.error);

File: deploy_pages.py
#!/usr/bin/env python3
"""Deploy Pages using Cloudflare API"""
import requests
import zipfile
import os
from pathlib import Path

API_TOKEN = "cfut_U9fdeBzWaVKBK83pfPZAtmufVQnJqtCTKLQj53UFad7fbeca"
ACCOUNT_ID = "dfbee5c2a5706a81bc04675499c933d4"
PROJECT_NAME = "upay-client-kc"

def create_zip(source_dir, output_path):
    """Create zip file from directory"""
    with zipfile.ZipFile(output_path, 'w', zipfile.ZIP_DEFLATED) as zipf:
        for root, dirs, files in os.walk(source_dir):
            for file in files:
                file_path = os.path.join(root, file)
                arcname = os.path.relpath(file_path, source_dir)
                zipf.write(file_path, arcname)
    return output_path

def deploy():
    headers = {
        "Authorization": f"Bearer {API_TOKEN}"
    }
    
    # Create zip
    source = "UpayClient/_KC"
    zip_file = "kc-deploy-py.zip"
    
    print("Creating zip file...")
    create_zip(source, zip_file)
    file_size = os.path.getsize(zip_file) / 1024
    print(f"Created: {zip_file} ({file_size:.2f} KB)")
    
    # Get upload token
    print("\nGetting upload token...")
    url = f"https://api.cloudflare.com/client/v4/accounts/{ACCOUNT_ID}/pages/projects/{PROJECT_NAME}/upload-token"
    
    try:
        resp = requests.get(url, headers=headers, timeout=30)
        data = resp.json()
        
        if data.get("success"):
            upload_token = data["result"]["upload_token"]
            upload_url = data["result"]["upload_url"]
            
            print(f"Upload URL: {upload_url[:60]}...")
            
            # Upload file
            print("\nUploading...")
            with open(zip_file, 'rb') as f:
                files = {'file': (zip_file, f, 'application/zip')}
                upload_headers = {
                    "Authorization": f"Bearer {API_TOKEN}"
                }
                upload_resp = requests.post(upload_url, headers=upload_headers, files=files, timeout=60)
            
            print(f"Upload status: {upload_resp.status_code}")
            print(f"Response: {upload_resp.text[:500]}")
            
            if upload_resp.status_code in [200, 201]:
                print("\n✅ Deployment successful!")
                print(f"URL: https://{PROJECT_NAME}.pages.dev")
            else:
                print(f"\n❌ Upload failed: {upload_resp.status_code}")
        else:
            print(f"❌ Failed to get token: {data}")
            
    except Exception as e:
        print(f"❌ Error: {e}")

if __name__ == "__main__":
    deploy()
